modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/ast/ArtifactAstUtils.java

```diff
@@ -107,43 +146,106 @@ public static ArtifactAst parseAndBuildAppExtensionModel(String[] configResource
+              .addParameter(MULE_SDK_EXTENSION_NAME_PROPERTY_NAME, artifactId)
               .build()));
     } else {
-      logModelNotGenerated("Mule ExtensionModelLoader not found", muleContext);
+      logModelNotGenerated("Mule ExtensionModelLoader not found", artifactId);
       return empty();
     }
   }
 
+  /**
+   * Handles a {@link ValidationResult} uniformly. Warnings are logged, errors raise exceptions.
+   *
+   * @param validationResult The {@link ValidationResult} obtained using an {@link ArtifactAstValidator}.
+   * @throws ConfigurationException If there was any result item with {@link ERROR} level.
+   */
+  public static void handleValidationResult(ValidationResult validationResult)
+      throws ConfigurationException {
+    final Collection<ValidationResultItem> items = validationResult.getItems();
+
+    // Logs warnings if any.
+    items.stream()
+        .filter(v -> v.getValidation().getLevel().equals(WARN))
+        .forEach(v -> LOGGER.warn(validationResultItemToString(v)));
+
+    // If there are any errors, throws.
+    final boolean hasErrors = items.stream().anyMatch(v -> v.getValidation().getLevel().equals(ERROR));
+    if (hasErrors) {
+      throw new ConfigurationException(createStaticMessage(validationResult.getItems()
+          .stream()
+          .map(ArtifactAstUtils::validationResultItemToString)
+          .collect(joining(lineSeparator()))));
+    }
+  }
+
+  private static String validationResultItemToString(ValidationResultItem v) {
+    return v.getComponents().stream()
+        .map(component -> component.getMetadata().getFileName().orElse("unknown") + ":"
+            + component.getMetadata().getStartLine().orElse(-1))
+        .collect(joining("; ", "[", "]")) + ": " + v.getMessage();
+  }
+
   private static ArtifactAst doParseArtifactIntoAst(String[] configResources,
                                                     AstXmlParserSupplier parserSupplier,
                                                     Set<ExtensionModel> extensions,
-                                                    boolean disableValidations) {
-    try {
-      return parserSupplier.getParser(extensions, disableValidations).parse(loadConfigResources(configResources));
-    } catch (Exception e) {
-      throw new MuleRuntimeException(e);
+                                                    boolean disableValidations)
+      throws ConfigurationException {
+    return parserSupplier.getParser(extensions, disableValidations).parse(loadConfigResources(configResources));
+  }
+
+  private static boolean definesOwnExtensionModel(ArtifactAst artifactAst) {
+    ArtifactType artifactType = artifactAst.getArtifactType();
+    switch (artifactType) {
+      case APPLICATION:
+        return artifactAst.topLevelComponentsStream()
+            .anyMatch(component -> APPLICATION_COMPONENT_TYPES.contains(component.getComponentType()));
```

Comment:
rename `APPLICATION_COMPONENT_TYPES` to something more like `MULE_EXTENSION_COMPONENT_TYPES`

Reply:
renamed to `REUSABLE_COMPONENT_TYPES` and added a TODO to move it somewhere else.